sdk/keyvault/Azure.Security.KeyVault.Shared/src/KeyVaultPipeline.cs

```diff
@@ -55,9 +55,10 @@ public Uri CreateFirstPageUri(string path, params ValueTuple<string, string>[] q
             return firstPage.ToUri();
         }
 
-        public Request CreateRequest(RequestMethod method, Uri uri, bool appendApiVersion)
+        public HttpMessage CreateRequest(RequestMethod method, ResponseClassifier classifier, Uri uri, bool appendApiVersion)
         {
-            Request request = _pipeline.CreateRequest();
+            HttpMessage message = _pipeline.CreateMessage(null, classifier);
+            Request request = message.Request;
```

Comment:
This approach LGTM.  One comment is that if you ever wanted to use other functionality available on RequestContext, or expose those APIs to callers, using RequestContext in the pipeline.CreateMessage() call rather than passing the classifier directly would give you that ability in the future.  But I don't think it's a huge deal if it's considerably more work.

Reply:
That's a good idea, but for now I'll add a comment as a design idea. For now, it'd just be an unnecessary allocation.